contracts/merkle/SuperMerkleAccess.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.7;
+
+import "../base/Sweepable.sol";
+
+/**
+  @title A merkle distributor.
+  @author Qazawat Zirak
+
+  This contract replaces the traditional whitelists by using a merkle 
+  tree, storing the root on-chain instead of all the addressses. The 
+  merkle tree alongside the whitelist is kept off-chain for lookups and 
+  creating proofs to validate a claim.
+  This code is inspired by and modified from incredible work from RicMoo.
+  https://github.com/ricmoo/ethers-airdrop/blob/master/AirDropToken.sol
+
+  October 12th, 2021.
+*/
+contract SuperMerkleAccess is Sweepable {
+
+  /// The public identifier for the right to set a root for a round.
+  bytes32 public constant SET_ROUND_ROOT = keccak256("SET_ROUND_ROOT");
+
+  /** 
+    The base struct containing whitelist information.
+    @param merkleRoot the proof stored on chain to verify against.
+    @param startTime the start time of validity for the whitelist.
+    @param endTime the end time of validity for the whitelist.
+    @param rounds the number times the whitelsit has been set.
+
+    A 0 value for startTime and endTime represent time indepedence.
```

Comment:
I don't like allowing 0 to be used as a flag for "no expiration" when it comes to `endTime`. I want to force people to be explicit and pass something like `MaxUint256` along to represent the notion of a whitelist that does not expire.

Using 0 as a flag for "no start time" / "already started" on `startTime` is just the default behavior.

```suggestion
```